.github/workflows/test.yml

```diff
@@ -146,6 +160,8 @@ jobs:
             xcode: "15.4"
             test-destination-os: "17.5"
             scheme: "Sentry"
+            additional-test-plans:
+              - "Sentry_HTTPTransport"
 
           # iOS 17
           - runs-on: macos-14
```

Comment:
This PR highlights to me that we have a job matrix of N runs that run the Sentry unit tests, and one that runs SentrySwiftUI. I would prefer we extract that SentrySwiftUI run to its own job, then the remaining will all also run the HTTP transport tests, so we don't have to use a matrix variable for it, or the extra logic in the run step. If we combine that with using a new test target to run the HTTP tests, like I've previously done with the profiler tests, this could be greatly simplified. It also moves us in the right direction for modularizing the SDK: I think our unit tests need to be broken down into targets per component anyways.

Reply:
> I would prefer we extract that SentrySwiftUI run to its own job

I agree with this 👍 

> If we combine that with using a new test target to run the HTTP tests...

This would work for me too, I do not see much of a difference in two test plans vs two targets, except for less target configuration management as mentioned in the other comment.